staging/src/k8s.io/apiserver/pkg/endpoints/handlers/negotiation/negotiate.go

```diff
@@ -273,19 +300,29 @@ func NegotiateMediaTypeOptions(header string, accepted []AcceptedMediaType, endp
 	}
 
 	clauses := goautoneg.ParseAccept(header)
+	candidateMediaTypeList := []candidateMediaType{}
```

Comment:
make this a `var candidates []candidateMediaType` array so that an allocation is not required in all cases.